src/controllers/tasks/active.officers.details.controller.ts

```diff
@@ -0,0 +1,44 @@
+import { ActiveOfficerDetails } from "@companieshouse/api-sdk-node/dist/services/confirmation-statement";
+import { Session } from "@companieshouse/node-session-handler";
+import { NextFunction, Request, Response } from "express";
+import { getActiveOfficersDetailsData } from "../../services/active.officers.details.service";
+import { CORPORATE_DIRECTORS_PATH, CORPORATE_SECRETARIES_PATH, NATURAL_PERSON_DIRECTORS_PATH, NATURAL_PERSON_SECRETARIES_PATH } from "../../types/page.urls";
+import { urlUtils } from "../../utils/url";
+
+export const get = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const transactionId = urlUtils.getTransactionIdFromRequestParams(req);
+    const submissionId = urlUtils.getSubmissionIdFromRequestParams(req);
+    const session: Session = req.session as Session;
+    const officers: ActiveOfficerDetails[] = await getActiveOfficersDetailsData(session, transactionId, submissionId);
+    let secretary = false;
+    let corpSecretary = false;
+    let director = false;
+    let corpDirector = false;
+
+    officers.forEach(officer => {
+      if (officer.role === "SECRETARY" && !officer.isCorporate){
```

Comment:
also, the "SECRETARY" and "DIRECTOR" are excellent candidates to be defined in the constants.ts module , possibly as enums?